packages/babel-plugin-transform-typescript/src/index.ts

```diff
@@ -63,6 +63,8 @@ export default declare((api, opts) => {
     var { allowDeclareFields = false } = opts;
   }
 
+  let hasValueExports = false;
```

Comment:
I believe this could also be attached to `state` instead of closing over it

Reply:
We can either use a `WeakMap` from the program node to this variable, or attach it to `state`.

Closing over causes problems, because Babel doesn't reinstantiate the plugin for each file but only when the options change.

Btw, do we need both `hasValueExports`/`hasValueImports`, or can it be a single variable?

Reply:
> I prefer the WeakMap approach since it's similar to what we use in many other plugins.

Great idea 🙂

> Btw, do we need both hasValueExports/hasValueImports, or can it be a single boolean?

Nope, only one is necessary. I'll combine them.